### PR TITLE
Only call MergeChangeList if something has changed

### PR DESCRIPTION
**Pull request by Vatyx**
#### Description
Was causing a crash in Scavengers. We should only be merging changelists if something has changed.

### CODE REVIEW

File: SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp [hunk 1]
```diff
@@ -279,7 +279,11 @@ bool USpatialActorChannel::ReplicateActor()
 		const int32 HistoryIndex = i % FRepChangelistState::MAX_CHANGE_HISTORY;
 		FRepChangedHistory& HistoryItem = ChangelistState->ChangeHistory[HistoryIndex];
```
Comment (danielimprobable) [inline SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: This code looks very similar to the one below, could we refactor this out to a function?

Comment (Vatyx) [reply SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: Probably out of the scope of this PR. This is a quick bugfix for Scavengers.

Comment (danielimprobable) [reply SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: Could you create a ticket? I feel like it wouldn't take long to just touch this up though

Comment (Vatyx) [reply SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: Created ticket: https://improbableio.atlassian.net/browse/UNR-663

**Pull request merged**